src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/DictionaryLayoutNode.cs

```diff
@@ -55,20 +55,138 @@ private void Validate()
             get;
         }
 
-        public abstract ICollection<NativeLayoutVertexNode> GetTemplateEntries(NodeFactory factory);
+        public TypeSystemEntity OwningMethodOrType => _owningMethodOrType;
 
-        public abstract void EmitDictionaryData(ref ObjectDataBuilder builder, NodeFactory factory, GenericDictionaryNode dictionary);
+        /// <summary>
+        /// Gets a value indicating whether the slot assignment is determined at the node creation time.
+        /// </summary>
+        public abstract bool HasFixedSlots
+        {
+            get;
+        }
+
+        public virtual ICollection<NativeLayoutVertexNode> GetTemplateEntries(NodeFactory factory)
+        {
+            ArrayBuilder<NativeLayoutVertexNode> templateEntries = new ArrayBuilder<NativeLayoutVertexNode>();
+            foreach (var entry in Entries)
+            {
+                templateEntries.Add(entry.TemplateDictionaryNode(factory));
+            }
```

Comment:
Obviously this shouldn't be fixed in this change, but you might think about it for future work.